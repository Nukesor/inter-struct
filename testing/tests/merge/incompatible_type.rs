use inter_struct::prelude::*;

/// This should produce a compile errors, as the field types for `other` are different.
#[derive(InterStruct)]
#[merge("crate::MergeStruct")]
pub struct FromStruct {
    pub normal: String,
    pub optional: i32,
}

pub struct MergeStruct {
    pub normal: String,
    pub optional: Option<String>,
}

fn main() {}
