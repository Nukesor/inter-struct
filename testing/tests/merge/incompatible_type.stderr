error[E0308]: mismatched types
 --> tests/merge/incompatible_type.rs:4:10
  |
4 | #[derive(InterStruct)]
  |          ^^^^^^^^^^^- help: try using a conversion method: `.to_string()`
  |          |
  |          expected struct `String`, found `i32`
  |
  = note: this error originates in the derive macro `InterStruct` (in Nightly builds, run with -Z macro-backtrace for more info)
